### PR TITLE
Migrate azure-batch-compute to azure-json

### PR DESCRIPTION
**Pull request by conniey**
# Description

* Migrate azure-batch-compute to azure-json.  Related to #40614
* Fix live test failure by supporting TestMode.LIVE.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by conniey** [comment]
@skapur12 Heyo, can you check over this migration PR ? Thanks!

**Comment by weidongxu-microsoft** [comment]
There is a bump of typespec-java version and the commit ID in tsp-location.yaml
https://github.com/Azure/azure-sdk-for-java/pull/40992

Hope it won't cause problem on the migrate.

**Review by lmolkova — APPROVED** [review]

File: sdk/batch/microsoft-azure-batch/src/test/java/com/microsoft/azure/batch/BatchIntegrationTestBase.java [hunk 1]
```diff
@@ -87,6 +87,8 @@ private static void initTestMode() throws IOException {
                 testMode = TestBase.TestMode.RECORD;
             } else if (azureTestMode.equalsIgnoreCase("Playback")) {
                 testMode = TestBase.TestMode.PLAYBACK;
+            } else if (azureTestMode.equalsIgnoreCase("Live")) {
+                testMode = TestMode.RECORD;
```
Comment (billwert) [inline sdk/batch/microsoft-azure-batch/src/test/java/com/microsoft/azure/batch/BatchIntegrationTestBase.java:91]: not `TestMode.LIVE`?

File: sdk/batch/azure-compute-batch/src/main/java/com/azure/compute/batch/models/CloudServiceConfiguration.java [hunk 2]
```diff
@@ -94,4 +95,56 @@ public CloudServiceConfiguration setOsVersion(String osVersion) {
...
+                    family = reader.getString();
+                } else if ("osVersion".equals(fieldName)) {
+                    version = reader.getString();
+                } else {
+                    reader.skipChildren();
+                }
+            }
+
+            if (family != null) {
+                return new CloudServiceConfiguration(family).setOsVersion(version);
```
Comment (billwert) [inline sdk/batch/azure-compute-batch/src/main/java/com/azure/compute/batch/models/CloudServiceConfiguration.java:144]: Will this be a potential regression? What would have happened previously if a json payload somehow didn't have a family?

File: sdk/batch/microsoft-azure-batch/src/test/java/com/microsoft/azure/batch/BatchIntegrationTestBase.java [hunk 3]
```diff
@@ -87,6 +87,8 @@ private static void initTestMode() throws IOException {
...
             } else {
```
Comment (srnagar) [inline sdk/batch/microsoft-azure-batch/src/test/java/com/microsoft/azure/batch/BatchIntegrationTestBase.java:92]: Unrelated to this PR: If the env var is not a known value, we should fallback to PLAYBACK mode. This is the behavior in azure-core-test as well. So, we should have consistent behavior here.

**Review by srnagar — APPROVED** [review]

File: sdk/batch/azure-compute-batch/src/main/java/com/azure/compute/batch/models/CloudServiceConfiguration.java (hunk 2 above)
Comment (conniey) [reply sdk/batch/azure-compute-batch/src/main/java/com/azure/compute/batch/models/CloudServiceConfiguration.java:144]: Urgh. Good find! :(

File: sdk/batch/microsoft-azure-batch/src/test/java/com/microsoft/azure/batch/BatchIntegrationTestBase.java (hunk 1 above)
Comment (conniey) [reply sdk/batch/microsoft-azure-batch/src/test/java/com/microsoft/azure/batch/BatchIntegrationTestBase.java:91]: This is another enumeration of the same name TestMode but it only supports RECORD and PLAYBACK.

**Review by billwert — APPROVED** [review]

**Pull request merged**